dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java

```diff
@@ -342,6 +379,156 @@ public IdentifierStripedLock getIdentifierStripedLock(){
+                    return null != SubmitterConfigCreatorBuilder.this.threadFactory?
+                            SubmitterConfigCreatorBuilder.this.threadFactory: SubmitterConfigCreator.super.getDefaultThreadFactory();
+                }
+
+                @Override
+                public RejectedExecutionHandler getRejectedExecutionHandler() {
+                    return null != SubmitterConfigCreatorBuilder.this.rejectedExecutionHandler?
+                            SubmitterConfigCreatorBuilder.this.rejectedExecutionHandler: SubmitterConfigCreator.super.getRejectedExecutionHandler();
+                }
+
+                @Override
+                public boolean getAllowCoreThreadTimeOut() {
+                    return null != SubmitterConfigCreatorBuilder.this.allowCoreThreadTimeOut?
+                            SubmitterConfigCreatorBuilder.this.allowCoreThreadTimeOut: SubmitterConfigCreator.super.getAllowCoreThreadTimeOut();
+                }
+
+                @Override
+                public int getPoolSize() {
+                    return null != SubmitterConfigCreatorBuilder.this.poolSize?
+                            SubmitterConfigCreatorBuilder.this.poolSize: SubmitterConfigCreator.super.getPoolSize();
+                }
+
+                @Override
+                public int getMaxPoolSize() {
+                    return null != SubmitterConfigCreatorBuilder.this.maxPoolSize?
+                            SubmitterConfigCreatorBuilder.this.maxPoolSize: SubmitterConfigCreator.super.getMaxPoolSize();
+                }
+
+                @Override
+                public long getKeepAliveMillis() {
+                    return null != SubmitterConfigCreatorBuilder.this.keepAliveMillis?
+                            SubmitterConfigCreatorBuilder.this.keepAliveMillis: SubmitterConfigCreator.super.getKeepAliveMillis();
+                }
+
+                @Override
+                public int getQueueCapacity() {
+                    return null != SubmitterConfigCreatorBuilder.this.queueCapacity?
+                            SubmitterConfigCreatorBuilder.this.queueCapacity: SubmitterConfigCreator.super.getQueueCapacity();
+                }
+            };
+        }
+    }
+
+    /**
+     * In case you want to configure by code.
+     */
+    public interface SubmitterConfigCreator {
+
+        /**
+         * Returns "Executors.defaultThreadFactory()"
+         * @return ThreadFactory
+         */
+        default ThreadFactory getDefaultThreadFactory () {
+            return Executors.defaultThreadFactory();
+        }
+
+        /**
+         * Returns AbortPolicy
+         * @return RejectedExecutionHandler
+         */
+        default RejectedExecutionHandler getRejectedExecutionHandler() {
+            return new ThreadPoolExecutor.AbortPolicy();
+        }
+
+        /**
+         * By default does not allows allow core time out
+         * @return boolean
+         */
+        default boolean getAllowCoreThreadTimeOut() {
```

Comment:
Codacy found an issue: [A getX() method which returns a boolean should be named isX()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6265363)